entries/serializers.py

```diff
@@ -27,13 +27,41 @@ class Meta:
             "updated_at",
         ]
 
-    def validate_plio(self, plio):
-        """Ensure that a session is created only for a published plio"""
-        if plio.status == "draft":
+    def validate(self, data):
+        """
+        All validation checks happen in this method. Currently these checks are defined
+        - A session can only be created for a published plio
+        - `is_first` is set properly depending on if a session already exists or not
+        """
+        # import ipdb; ipdb.set_trace()
+        # validate plio's status
+        if data["plio"].status == "draft":
             raise serializers.ValidationError(
                 "A session can only be created for a published plio"
             )
-        return plio
+
+        # set `is_first` properly
+        last_session = (
+            Session.objects.filter(plio_id=data["plio"].id)
+            .filter(user_id=data["user"].id)
+            .first()
+        )
+
+        if self.context["view"].action == "create":
+            # while creating a session, check if any session already exists
+            if last_session:
+                data["is_first"] = False
+            else:
+                data["is_first"] = True
+        elif self.context["view"].action == "update":
+            # while updating a session, check if the session being updated
+            # is the very first session
+            if f"{last_session.id}" == self.context["view"].kwargs["pk"]:
```

Comment:
can we use `id` instead of `pk` - just like we changed `pk` with `plio` to `uuid`- better to just check `id` if it is available?